### PR TITLE
[IMP] l10n_th: localization improvement

### PR DESCRIPTION
**Pull request by nni-odoo**
Description of the issue/feature this PR addresses:

Thailand localization improvement that includes:
- Customized Thai reports for reports, bank payments
- Thailand States initialized
- Template for translation of Thai phrases
- 2 new tax reports and adding them as tax grids on different taxes

2879718, 2895669

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_th/views/report_invoice.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH'" >
+                <span t-field="o.l10n_th_company_info" />
+                <span t-field="o.l10n_th_branch_number" />
+            </t>
+        </xpath>
+        <xpath expr="//h2" position="replace">
```
Comment (fja-odoo) [inline addons/l10n_th/views/report_invoice.xml:10]: Isn't there a better way to do this without replacing the whole h2 ?

File: addons/l10n_th/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    vat = fields.Char(related="partner_id.vat", string="Tax ID")
```
[the remark this answers is not in the document]
Comment (nni-odoo) [reply addons/l10n_th/models/account_move.py:10]: we don't have l10n_th_vat field on res.partner

File: addons/l10n_th/views/report_invoice.xml (hunk 1 above)
Comment (nni-odoo) [reply addons/l10n_th/views/report_invoice.xml:10]: I think we would need to change the attribute of every line inside h2 anyway, seems like replacing it altogether is the more efficient way

File: addons/l10n_th/models/account_move.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fja-odoo) [reply addons/l10n_th/models/account_move.py:10]: I mean the name of the field

File: addons/l10n_th/views/report_invoice.xml (hunk 1 above)
Comment (fja-odoo) [reply addons/l10n_th/views/report_invoice.xml:10]: Can't you add your h2 before with a t-if use_custom_l10n_header
And add a t-else on the existing h2

File: addons/l10n_th/views/report_invoice.xml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        </xpath>
+        <xpath expr="//t[@t-call='account.document_tax_totals']" position="replace">
+            <t t-if="o.l10n_th_custom_header and o.company_id.account_fiscal_country_id.code == 'TH' and o.debit_origin_id">
+                <t t-call="l10n_th.document_tax_totals"/>
+            </t>
+            <t t-else="">
+                <t t-call="account.document_tax_totals"/>
+            </t>
+        </xpath>
+    </template>
```
Comment (fja-odoo) [inline addons/l10n_th/views/report_invoice.xml:32]: separate the 2 templates with an empty line

**Pull request closed without merging**